plugins/woocommerce/client/blocks/assets/js/blocks/add-to-cart-with-options/components/upgrade-product-image-gallery/index.tsx

```diff
@@ -0,0 +1,73 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useState, useEffect } from '@wordpress/element';
+import { createBlock, BlockInstance } from '@wordpress/blocks';
+import { dispatch, useSelect } from '@wordpress/data';
+import { UpgradeDowngradeNotice } from '@woocommerce/editor-components/upgrade-downgrade-notice';
+import { findBlock } from '@woocommerce/utils';
+
+const upgradeToBlockifiedProductGallery = (
+	productImageGalleryBlockClientId: string
+) => {
+	const newBlock = createBlock( 'woocommerce/product-gallery' );
+
+	dispatch( 'core/block-editor' ).replaceBlock(
+		productImageGalleryBlockClientId,
+		newBlock
+	);
+
+	return true;
+};
+
+export const UpgradeProductImageGallery = () => {
+	const [ productImageGalleryBlock, setProductImageGalleryBlock ] =
+		useState< BlockInstance | null >( null );
+
+	const getBlocks = useSelect( ( select ) => {
+		return select( 'core/block-editor' ).getBlocks;
+	}, [] );
```

Comment:
Unsure about this here… instead of using the `useSelect` hook, can't you just import `select` itself at this point, since you're only storing a reference to the function?

Speaking of which, since the depArray below is only `getBlocks` and `setProductImageGalleryBlock`, this won't react to changes in the editor, no? I am not sure in which case this might break when the block change, but it _feels_ there might be a situation in which this breaks.

Reply:
Yea, I think my second paragraph is probably just an overly defensive concern. In the real world, it might now happen. However, I'm still not sure why not importing `select` instead of getting a reference to `getBlock` at every re-render.